roles/postgres_replicant/tasks/main.yml

```diff
@@ -18,9 +18,19 @@
     - database
     - replication
 
+- name: check that replication connection is possible
+  become: yes
+  become_user: postgres
+  command: psql 'dbname=postgres replication=database host={{ hostvars[groups.database[0]].ansible_default_ipv4.address }} user={{ replicator_db_user|default(default_replicator_db_user) }}' -c 'IDENTIFY_SYSTEM;'
+  register: replication_connection
+  tags:
+    - database
+    - replication
+
 - when:
     - "'f' in replication_running.stdout"
     - not recoveryconf_file.stat.exists
+    - "'systemid' in replication_connection.stdout"
```

Comment:
You'll need to add this to the next task as well, no? Because if this doesn't run and the following task does, the `recovery.conf` will be created. This means that on next run, this would fail the condition for another reason, which is probably not the intended case.